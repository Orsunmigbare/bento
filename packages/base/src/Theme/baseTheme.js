import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
