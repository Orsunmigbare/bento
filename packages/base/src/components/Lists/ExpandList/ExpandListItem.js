import React, { Component } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { BounceIn } from "@staccx/animations"

const Caret = ({ className }) => (
  <svg className={className} viewBox="0 0 24 16" width="18">
    <path d="M3.81 0a.54.54 0 0 0-.38.16L.16 3.47a.56.56 0 0 0 0 .78l11.45 11.59a.54.54 0 0 0 .78 0L23.84 4.25a.56.56 0 0 0 0-.78L20.57.17a.54.54 0 0 0-.77 0L12 8.04 4.2.16A.54.54 0 0 0 3.81 0z" />
  </svg>
)
Caret.propTypes = { className: PropTypes.string }
Caret.defaultProps = { className: "" }

class ExpandListItem extends Component {
  constructor(props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
    this.state = {
      isExpanded: this.props.expanded
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.expanded !== undefined) {
      this.setState({ isExpanded: nextProps.expanded })
    }
  }

  handleChange(event) {
    this.setState({
      isExpanded: !this.state.isExpanded
    })
    if (this.props.onClick) {
      this.props.onClick(event)
    }
  }
  render() {
    const {
      title,
      children,
      className,
      flush,
      onClick,
      ...otherProps
    } = this.props
    return (
      <ExpandItem className={className} {...otherProps}>
        <ExpandBtn
          isExpanded={this.state.isExpanded}
          onClick={this.handleChange}
          aria-expanded={this.state.isExpanded}
          aria-controls={title}
          id={title + "2"}
        >
          {title} <ExpandIcon isExpanded={this.state.isExpanded} />
        </ExpandBtn>
        {this.state.isExpanded && (
          <ExpandedItem flush={flush} id={title} aria-labelledby={title + "2"}>
            {children}
          </ExpandedItem>
        )}
      </ExpandItem>
    )
  }
}

const ExpandItem = styled.li`
  border-bottom: 1px solid ${p => p.theme.color.line};
  position: relative;

  &:last-child {
    border-bottom-width: 0;
  }
`

const ExpandIcon = styled(Caret)`
  position: absolute;
  top: 16px;
  right: ${p => p.theme.spacing.small};
  fill: ${p => p.theme.color.line};
  transition: transform 0.3s ease-out;
  transform: ${p => (p.isExpanded ? "rotate(180deg)" : "rotate(0)")};
`

const ExpandBtn = styled.button`
  background: transparent;
  border-width: 0;
  color: ${p => p.theme.color.text};
  font-size: ${p => p.theme.font.size.base};
  font-weight: ${p => (p.isExpanded ? "bold" : "normal")};
  cursor: pointer;
  padding: ${p => p.theme.spacing.small} ${p => p.theme.spacing.large}
    ${p => p.theme.spacing.small} ${p => p.theme.spacing.medium};
  width: 100%;
  text-align: left;

  &:hover,
  &:focus {
    color: ${p => p.theme.color.black};
    outline: none;

    ${ExpandIcon} {
      fill: ${p => p.theme.color.primary};
    }
  }
`

const ExpandedItem = styled.div`
  opacity: 0;
  animation: 0.4s ${BounceIn} 0.05s ease-out forwards 1;
  padding: ${p => (p.flush ? 0 : p.theme.spacing.small)}
    ${p => (p.flush ? 0 : p.theme.spacing.medium)}
    ${p => (p.flush ? 0 : p.theme.spacing.medium)};
  max-width: ${p => (p.flush ? "auto" : "540px")};
  margin: auto;
  line-height: 1.8;
`

ExpandListItem.defaultProps = {
  expanded: false,
  className: "",
  flush: false
}

ExpandListItem.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.array
  ]).isRequired,
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.array
  ]).isRequired,
  className: PropTypes.string,
  expanded: PropTypes.bool,
  flush: PropTypes.bool
}

export default ExpandListItem
