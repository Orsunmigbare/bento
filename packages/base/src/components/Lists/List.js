import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
`

const List = ({ children, className }) => (
  <ListUl className={className}>{children}</ListUl>
)

List.defaultProps = {
  className: ""
}

List.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element
  ]).isRequired,
  className: PropTypes.string
}

export default List
