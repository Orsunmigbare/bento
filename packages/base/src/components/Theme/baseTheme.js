export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
