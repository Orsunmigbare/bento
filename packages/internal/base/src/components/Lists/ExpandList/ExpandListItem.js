import React, { Component } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { BounceIn } from "@staccx/animations"
import Caret from "../../Icons/Caret"
import {
  spacing,
  font,
  color,
  themify,
  ThemeComponent,
  themeProps
} from "@staccx/theme"

const ExpandButton = ({ title, isExpanded, ...props }) => (
  <ExpandBtn isExpanded={isExpanded} {...props}>
    {title} <ExpandIcon isExpanded={isExpanded} />
  </ExpandBtn>
)

export const COMPONENT_EXPAND_LIST_ITEM_BTN = "COMPONENT_EXPAND_LIST_ITEM_BTN"

const BtnComponent = ({ ...props }) => (
  <ThemeComponent
    tagName={COMPONENT_EXPAND_LIST_ITEM_BTN}
    fallback={ExpandButton}
    {...props}
  />
)
class ExpandListItem extends Component {
  constructor(props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
    this.state = {
      isExpanded: this.props.expanded
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.expanded !== undefined) {
      this.setState({ isExpanded: nextProps.expanded })
    }
  }

  handleChange(event) {
    this.setState({
      isExpanded: !this.state.isExpanded
    })
    if (this.props.onClick) {
      this.props.onClick(event)
    }
  }
  render() {
    const {
      title,
      children,
      className,
      flush,
      onClick,
      variant,
      ...otherProps
    } = this.props
    const { isExpanded } = this.state
    return (
      <ExpandItem className={className} variant={variant} {...otherProps}>
        <BtnComponent
          {...otherProps}
          title={title}
          isExpanded={isExpanded}
          onClick={this.handleChange}
          aria-expanded={isExpanded}
          aria-controls={title}
          id={title + "2"}
          variant={variant}
        />
        {isExpanded && (
          <ExpandedItem
            flush={flush}
            id={title}
            aria-labelledby={title + "2"}
            variant={variant}
          >
            {children}
          </ExpandedItem>
        )}
      </ExpandItem>
    )
  }
}

export const EXPAND_LIST_ITEM = "EXPAND_LIST_ITEM"
const ExpandItem = styled.li`
  border-bottom: 1px solid ${color.line};
  position: relative;

  &:last-child {
    border-bottom-width: 0;
  }
  ${themify(EXPAND_LIST_ITEM)};
`

export const COMPONENT_EXPAND_LIST_ICON = "COMPONENT_EXPAND_LIST_ICON"
const IconComponent = ({ ...props }) => (
  <ThemeComponent
    tagName={COMPONENT_EXPAND_LIST_ICON}
    fallback={Caret}
    {...props}
  />
)
export const EXPAND_LIST_ICON = "EXPAND_LIST_ICON"
const ExpandIcon = styled(IconComponent)`
  position: absolute;
  right: ${spacing.small()};
  fill: ${color.wcag};
  transition: transform 0.3s ease-out;
  transform: ${p => (p.isExpanded ? "rotate(180deg)" : "rotate(0)")};
  ${themify(EXPAND_LIST_ICON)};
`

export const EXPAND_LIST_BUTTON = "EXPAND_LIST_BUTTON"
const ExpandBtn = styled.button`
  background: transparent;
  border-width: 0;
  color: ${color.text};
  font-size: ${font.base};
  font-weight: ${p => (p.isExpanded ? "bold" : "normal")};
  cursor: pointer;
  padding: ${spacing.small()} ${spacing.large()} ${spacing.small()}
    ${spacing.medium()};
  width: 100%;
  text-align: left;
  display: flex;
  justify-content: space-between;
  align-items: center;

  &:hover,
  &:focus {
    color: ${color.black};
    outline: none;

    ${ExpandIcon} {
      fill: ${color.primary};
    }
  }
  ${themify(EXPAND_LIST_BUTTON)};
`

export const EXPAND_LIST_EXPANDED_ITEM = "EXPAND_LIST_EXPANDED_ITEM"
const ExpandedItem = styled.div`
  opacity: 0;
  animation: 0.4s ${BounceIn} 0.05s ease-out forwards 1;
  padding: ${p => (p.flush ? 0 : spacing.small())}
    ${p => (p.flush ? 0 : spacing.medium())}
    ${p => (p.flush ? 0 : spacing.medium())};
  max-width: ${p => (p.flush ? "auto" : "540px")};
  margin: auto;
  line-height: 1.8;
  ${themify(EXPAND_LIST_EXPANDED_ITEM)};
`

ExpandListItem.defaultProps = {
  expanded: false,
  className: "",
  flush: false
}

ExpandListItem.propTypes = {
  children: themeProps.children.isRequired,
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.array
  ]).isRequired,
  className: PropTypes.string,
  expanded: PropTypes.bool,
  flush: PropTypes.bool
}

export default ExpandListItem
