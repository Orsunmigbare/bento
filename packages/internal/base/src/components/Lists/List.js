import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { themify, themeProps } from "@staccx/theme"

const List = ({ children, className, ...restProps }) => (
  <ListUl className={className} {...restProps}>
    {children}
  </ListUl>
)

export const LIST = "LIST"

export const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
  ${themify(LIST)};
`

List.defaultProps = {
  className: ""
}

List.propTypes = {
  children: themeProps.children.isRequired,
  className: PropTypes.string
}

export default List
