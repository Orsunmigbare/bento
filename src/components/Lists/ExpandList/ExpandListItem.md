```jsx
<List>
  <ExpandListItem title="Hvor lang tid tar det før jeg får svar på søknaden og hvorfor er denne teksten så lang som den faktisk er og la oss se hvordan dette ser ut?">
    Du får umiddelbart svar på søknaden og kan se hvor mye du får i lån.
  </ExpandListItem>
  <ExpandListItem title="Hvor lang tid tar det før lånet blir utbetalt?">
    Normalt vil du ha pengene på din konto i annen bank en til to virkedager
    etter vi har mottatt søknad og godkjent all dokumentasjon. Om du ønsker
    utbetaling på dagen kontakter du vårt kundesenter.
  </ExpandListItem>
</List>
```
