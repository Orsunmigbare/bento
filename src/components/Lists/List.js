import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { applyVariants, commonPropTypes, themePropTypes } from "../../theming"

const List = ({ children, className, ordered, ...restProps }) => {
  if (!ordered) {
    return (
      <ListUl className={className} {...restProps}>
        {children}
      </ListUl>
    )
  }
  return (
    <ListOl className={className} {...restProps}>
      {children}
    </ListOl>
  )
}

List.themeProps = {
  list: {
    name: "LIST",
    description: "",
    type: themePropTypes.style
  }
}
export const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
  ${applyVariants(List.themeProps.list)};
`

export const ListOl = ListUl.withComponent("ol")

List.defaultProps = {
  className: "",
  ordered: false
}

List.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string,
  /**
   * Prop to set ordered or unordered List (ul/ol)
   */
  ordered: PropTypes.bool
}

export default List
