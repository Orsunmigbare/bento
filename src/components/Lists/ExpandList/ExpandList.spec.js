import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import ExpandListItem from "./ExpandListItem"
import List from "../List.js"

describe("ExpandListItem", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <ExpandListItem>
                Du får umiddelbart svar på søknaden og kan se hvor mye du får i
                lån
              </ExpandListItem>
              <ExpandListItem>
                Normalt vil du ha pengene på din konto i annen bank en til to
                virkedager etter vi har mottatt søknad og godkjent all
                dokumentasjon. Om du ønsker utbetaling på dagen kontakter du
                vårt kundesenter.
              </ExpandListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Title", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <ExpandListItem title="Hvor lang tid tar det før jeg får svar på søknaden?">
                Du får umiddelbart svar på søknaden og kan se hvor mye du får i
                lån
              </ExpandListItem>
              <ExpandListItem title="Hvor lang tid tar det før lånet blir utbetalt?">
                Normalt vil du ha pengene på din konto i annen bank en til to
                virkedager etter vi har mottatt søknad og godkjent all
                dokumentasjon. Om du ønsker utbetaling på dagen kontakter du
                vårt kundesenter.
              </ExpandListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("expanded", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <ExpandListItem expanded>
                Du får umiddelbart svar på søknaden og kan se hvor mye du får i
                lån
              </ExpandListItem>
              <ExpandListItem expanded>
                Normalt vil du ha pengene på din konto i annen bank en til to
                virkedager etter vi har mottatt søknad og godkjent all
                dokumentasjon. Om du ønsker utbetaling på dagen kontakter du
                vårt kundesenter.
              </ExpandListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("flush", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <ExpandListItem flush>
                Du får umiddelbart svar på søknaden og kan se hvor mye du får i
                lån
              </ExpandListItem>
              <ExpandListItem flush>
                Normalt vil du ha pengene på din konto i annen bank en til to
                virkedager etter vi har mottatt søknad og godkjent all
                dokumentasjon. Om du ønsker utbetaling på dagen kontakter du
                vårt kundesenter.
              </ExpandListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("All", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <ExpandListItem flush expanded title="test">
                Du får umiddelbart svar på søknaden og kan se hvor mye du får i
                lån
              </ExpandListItem>
              <ExpandListItem flush expanded title="test">
                Normalt vil du ha pengene på din konto i annen bank en til to
                virkedager etter vi har mottatt søknad og godkjent all
                dokumentasjon. Om du ønsker utbetaling på dagen kontakter du
                vårt kundesenter.
              </ExpandListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
