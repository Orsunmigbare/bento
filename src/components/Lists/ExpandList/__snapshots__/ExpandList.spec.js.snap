// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandListItem Snapshots All 1`] = `
<ul
  className="sc-AykKG iaaMfj"
>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-controls="test"
      aria-expanded={true}
      className="sc-AykKE oXQE"
      id="test2"
      onClick={[Function]}
    >
      test
       
      <svg
        className="sc-AykKD kjgsMs"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
    <div
      aria-labelledby="test2"
      className="sc-AykKF ewHXzn"
      id="test"
    >
      Du får umiddelbart svar på søknaden og kan se hvor mye du får i lån
    </div>
  </li>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-controls="test"
      aria-expanded={true}
      className="sc-AykKE oXQE"
      id="test2"
      onClick={[Function]}
    >
      test
       
      <svg
        className="sc-AykKD kjgsMs"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
    <div
      aria-labelledby="test2"
      className="sc-AykKF ewHXzn"
      id="test"
    >
      Normalt vil du ha pengene på din konto i annen bank en til to virkedager etter vi har mottatt søknad og godkjent all dokumentasjon. Om du ønsker utbetaling på dagen kontakter du vårt kundesenter.
    </div>
  </li>
</ul>
`;

exports[`ExpandListItem Snapshots No props 1`] = `
<ul
  className="sc-AykKG iaaMfj"
>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-expanded={false}
      className="sc-AykKE dkJeRc"
      id="undefined2"
      onClick={[Function]}
    >
       
      <svg
        className="sc-AykKD fCrqUz"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
  </li>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-expanded={false}
      className="sc-AykKE dkJeRc"
      id="undefined2"
      onClick={[Function]}
    >
       
      <svg
        className="sc-AykKD fCrqUz"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
  </li>
</ul>
`;

exports[`ExpandListItem Snapshots Title 1`] = `
<ul
  className="sc-AykKG iaaMfj"
>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-controls="Hvor lang tid tar det før jeg får svar på søknaden?"
      aria-expanded={false}
      className="sc-AykKE dkJeRc"
      id="Hvor lang tid tar det før jeg får svar på søknaden?2"
      onClick={[Function]}
    >
      Hvor lang tid tar det før jeg får svar på søknaden?
       
      <svg
        className="sc-AykKD fCrqUz"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
  </li>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-controls="Hvor lang tid tar det før lånet blir utbetalt?"
      aria-expanded={false}
      className="sc-AykKE dkJeRc"
      id="Hvor lang tid tar det før lånet blir utbetalt?2"
      onClick={[Function]}
    >
      Hvor lang tid tar det før lånet blir utbetalt?
       
      <svg
        className="sc-AykKD fCrqUz"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
  </li>
</ul>
`;

exports[`ExpandListItem Snapshots expanded 1`] = `
<ul
  className="sc-AykKG iaaMfj"
>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-expanded={true}
      className="sc-AykKE oXQE"
      id="undefined2"
      onClick={[Function]}
    >
       
      <svg
        className="sc-AykKD kjgsMs"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
    <div
      aria-labelledby="undefined2"
      className="sc-AykKF hBKqNe"
    >
      Du får umiddelbart svar på søknaden og kan se hvor mye du får i lån
    </div>
  </li>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-expanded={true}
      className="sc-AykKE oXQE"
      id="undefined2"
      onClick={[Function]}
    >
       
      <svg
        className="sc-AykKD kjgsMs"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
    <div
      aria-labelledby="undefined2"
      className="sc-AykKF hBKqNe"
    >
      Normalt vil du ha pengene på din konto i annen bank en til to virkedager etter vi har mottatt søknad og godkjent all dokumentasjon. Om du ønsker utbetaling på dagen kontakter du vårt kundesenter.
    </div>
  </li>
</ul>
`;

exports[`ExpandListItem Snapshots flush 1`] = `
<ul
  className="sc-AykKG iaaMfj"
>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-expanded={false}
      className="sc-AykKE dkJeRc"
      id="undefined2"
      onClick={[Function]}
    >
       
      <svg
        className="sc-AykKD fCrqUz"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
  </li>
  <li
    className="sc-AykKC hWMszK"
  >
    <button
      aria-expanded={false}
      className="sc-AykKE dkJeRc"
      id="undefined2"
      onClick={[Function]}
    >
       
      <svg
        className="sc-AykKD fCrqUz"
        height="7"
        viewBox="0 0 33 20"
        width="12"
      >
        <path
          d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
        />
      </svg>
    </button>
  </li>
</ul>
`;
